lib/sinon/collection.js

```diff
@@ -46,7 +46,16 @@ var collection = {
     },
 
     resetHistory: function resetHistory() {
-        each(this, "resetHistory");
+        getFakes(this).forEach(function (fake) {
+            if (typeof fake.resetHistory === "function") {
+                fake.resetHistory();
+                return;
```

Comment:
I definitely agree with point number 2. I'm even working on a PR related to that. But point number one is a little misguided. 

1. Nothing is special about `else-if` in javascript. The logic is the same in every language that I can think of. You can always rewrite an else-if as a 3 statements instead of 2. It's a core tenant of Logic.
2. Splashing random returns around seems less explicit and it's obscuring the intent. Is the goal of this code to `return undefined`?
3. The most important difference to me is consistency.  We use else-if throughout the code. In my opinion, it reads better and is easier to reason about.

Another problem I foresee is future work. So let's say there was something I wanted to do no matter which fork of the `if` branch the code went down.. I'd have to "fix" your code first before I could accomplish that. In an `if-else` case, the change would be purely additive.

How would you feel adding a (pointless) `return` statement to the other if for consistency's sake?